plugins/module_utils/rds.py

```diff
@@ -83,7 +87,39 @@
 ]
 
 
-def get_rds_method_attribute(method_name, module):
+class AnsibleRDSError(AnsibleAWSError):
+    pass
+
+
+class RDSErrorHandler(AWSErrorHandler):
+    _CUSTOM_EXCEPTION = AnsibleRDSError
+
+    @classmethod
+    def _is_missing(cls):
+        return is_boto3_error_code(["DBInstanceNotFound", "DBSnapshotNotFound", "DBClusterNotFound"])
+
+
+@RDSErrorHandler.list_error_handler("describe db instances", [])
+@AWSRetry.jittered_backoff()
+def describe_db_instances(client, **params: Dict) -> List[Dict[str, Any]]:
+    paginator = client.get_paginator("describe_db_instances")
+    return paginator.paginate(**params).build_full_result()["DBInstances"]
+
+
+@RDSErrorHandler.list_error_handler("describe db snapshots", [])
+@AWSRetry.jittered_backoff()
+def describe_db_snapshots(client, **params: Dict) -> List[Dict]:
+    paginator = client.get_paginator("describe_db_snapshots")
+    return paginator.paginate(**params).build_full_result()["DBSnapshots"]
+
+
+@RDSErrorHandler.list_error_handler("list tags for resource", [])
+@AWSRetry.jittered_backoff()
+def list_tags_for_resource(client, resource_arn: str) -> List[Dict[str, str]]:
```

Comment:
I can look into that for a future PR

Reply:
There's a certain amount of code that can (in theory) be shared, unfortunately the AWS APIs are somewhat less than consistent with how they handle tagging.  Some of the common code (such as comparing "boto3" tag lists) has already been moved, but we may need to do something like having 3 or 4 variants that can be used.

- Some APIs have different names for the same "command"
- Some APIs overwrite all tags, others require an explicit deletion
- Some APIs return the tags as part of the main object (and don't need a list call)
- A handful of APIs use dict-style tags rather than list-of-dict

What we may need is something class based so we can configure the tagging "settings" on a per-API basis in a python module specifically for that API (ec2.py / rds.py) while still putting as much of the code into a common location as possible.  (Triggering all integration tests to add tagging support for a new API will just get annoying)